### ISSUE TITLE
Upload directly in the right panel

### ISSUE THREAD

**Issue by fabriceci** [servocoder/RichFilemanager#149]
A small proposal to improve ergonomics. Be able to drop files directly in the right panel zone.

**Comment by servocoder** [comment]
This may work, but we have to consider a lot of details. Move and drop files directly to the right panel (mixing them with exisiting files) is a bad idea, and it will be problematic due to technical reasons. There should be some compromise, I believe. For example, an upload panel may appear (slide down) at the top of the right panel when you drag files over it.

**Comment by servocoder** [comment]
Technical details:

Each item in the right panel is an instance of a sprcific JS class (ItemModel) with a set of properties and methods which are used and triggered in the view (html file). Files prepared for upload (which are not uploaded yet), can't have such properties and couldn't be added to the right panel before completely uploaded. At the same time you could see that the current upload model flow consists of few steps. You can see whether a file successfully added to upload queue and whether it is successfully uploaded or some error happens. And which files are failed in queue. I can't see a way how to provide all these features without some intermediate step (popup or panel), because we can add files directly to the right panel only after they are successfully uploaded.

**Comment by fabriceci** [comment]
Thanks for the details, like you suggested, the same panel that the current one, which appears automatically when you are going to drop files, is a very good solution!

**Comment by ShortBusKing** [comment]
I agree this is a great idea. Could it be possible to upload whole folders with their subfiles as well?
